latest/src/node_activation.rs

```diff
@@ -73,42 +73,40 @@ pub trait ContractStakeModule {
             );
 
             node_ids.push(node_id);
-            bls_keys_signatures.push(bls_key.to_vec());
-            bls_keys_signatures.push(self.node_config().get_node_signature(node_id).to_vec());
+            let bls_signature = self.node_config().get_node_signature(node_id);
+            bls_keys_signatures.push((bls_key, bls_signature).into());
 
             self.node_config()
                 .set_node_state(node_id, NodeState::PendingActivation);
         }
 
-        self.perform_stake_nodes(node_ids, bls_keys_signatures, amount_to_stake)
+        Ok(self.perform_stake_nodes(node_ids, bls_keys_signatures.into(), amount_to_stake))
     }
 
     fn perform_stake_nodes(
         &self,
         node_ids: Vec<usize>,
-        bls_keys_signatures: Vec<Vec<u8>>,
+        bls_keys_signatures: VarArgs<MultiArg2<BLSKey, BLSSignature>>,
         amount_to_stake: BigUint,
-    ) -> SCResult<()> {
+    ) -> AsyncCall<BigUint> {
         let num_nodes = node_ids.len();
         // send all stake to auction contract
         let auction_contract_addr = self.settings().get_auction_contract_address();
-        let auction_contract = contract_proxy!(self, &auction_contract_addr, Auction);
-        auction_contract.stake(
-            node_ids, // callback arg
-            num_nodes,
-            bls_keys_signatures.into(),
-            &amount_to_stake,
-        );
 
-        Ok(())
+        contract_call!(self, auction_contract_addr, AuctionProxy)
+            .with_token_transfer(TokenIdentifier::egld(), amount_to_stake)
+            .stake(num_nodes, bls_keys_signatures)
+            .async_call()
+            .with_callback(self.callbacks().auction_stake_callback(node_ids))
     }
 
     /// Only finalize activation if we got confirmation from the auction contract.
-    /// #[callback] can only be declared in lib.rs for the moment.
+    /// `#[callback]` also has be declared in lib.rs for the moment.
+    #[callback]
     fn auction_stake_callback(
         &self,
         node_ids: Vec<usize>, // #[callback_arg]
-        call_result: AsyncCallResult<VarArgs<BLSStatusMultiArg>>,
+        #[call_result] call_result: AsyncCallResult<VarArgs<BLSStatusMultiArg>>,
```

Comment:
Optional: Since this is a result, maybe use MultiResultVec<> instead of VarArgs<>